### PR TITLE
Create tests for book controller

### PR DESCRIPTION
**Pull request by silvitaelen**
## Summary

- Creating tests for routes index and show BookController.
- I added context for `Authenticated User` into `spec/support`.
- I added helper `parsed_response_helper.rb`.
- Excluded `spec/` to `Metrics/BlockLength` in `rubocop.yml`

## Trello Card

- https://trello.com/c/ky79dbRc

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -3,3 +3,7 @@ Documentation:
 
 LineLength:
   Max: 99
+
+Metrics/BlockLength:
+  Exclude:
+    - spec/**/*
```
Comment (Atixx) [inline .rubocop.yml]: rememeber to have a new line at the end of the file

File: spec/controllers/books_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  describe 'GET #index' do
+    context 'When fetching all the books' do
+      let!(:books) { create_list(:book, 20) }
+
+      before do
+        get :index
+      end
+
+      it 'responses with the books json' do
+        expected = ActiveModelSerializers::SerializableResource.new(
```
Comment (Atixx) [inline spec/controllers/books_controller_spec.rb]: no need to assign a variable

File: spec/controllers/books_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe BooksController, type: :controller do
+  include_context 'Authenticated User'
+
+  describe 'GET #index' do
+    context 'When fetching all the books' do
+      let!(:books) { create_list(:book, 20) }
```
Comment (Atixx) [inline spec/controllers/books_controller_spec.rb:10]: try to avoid let! (with bang)

File: spec/factories/rents.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :rent do
+    user { user_id }
```
Comment (Atixx) [inline spec/factories/rents.rb]: Have your factory be able to create all it needs when building the model, so if you just need a rent and you don't care about the user, you only call the rent factory, and it takes care of all the relations

**Review by Atixx — CHANGES_REQUESTED** [review]

File: .rubocop.yml (hunk 1 above)
Comment (silvitaelen) [reply .rubocop.yml]: ready!

File: spec/controllers/books_controller_spec.rb [hunk 5]
```diff
@@ -7,17 +7,16 @@
...
-      let!(:books) { create_list(:book, 20) }
+      let(:books) { create_list(:book, 20) }
 
       before do
         get :index
       end
 
       it 'responses with the books json' do
-        expected = ActiveModelSerializers::SerializableResource.new(
+        expect(response_body.to_json) == ActiveModelSerializers::SerializableResource.new(
```
Comment (Atixx) [inline spec/controllers/books_controller_spec.rb]: use `.to eq` like you did in the other example

File: spec/factories/rents.rb [hunk 6]
```diff
@@ -2,8 +2,8 @@
 
 FactoryBot.define do
   factory :rent do
-    user { user_id }
-    book { book_id }
+    user_id { create(:user).id }
```
Comment (Atixx) [inline spec/factories/rents.rb]: You don't need to specify id and do `user { create(:user) }`, so its a bit clearer.

**Review by Atixx — CHANGES_REQUESTED** [review]

File: spec/controllers/books_controller_spec.rb (hunk 5 above)
Comment (silvitaelen) [reply spec/controllers/books_controller_spec.rb]: when I deleted the variable 'expected', I threw this error and the '==' suggestion when running the tests
image

**Review by Atixx — APPROVED** [review]

**Pull request merged**